pytest_fixtures/core/contenthosts.py

```diff
@@ -180,16 +179,38 @@ def rex_contenthosts(request, module_org, target_sat, module_ak_with_cv):
 
 @pytest.fixture
 def katello_host_tools_tracer_host(rex_contenthost, target_sat):
-    """Install katello-host-tools-tracer, create custom
-    repositories on the host"""
+    """Install katello-host-tools-tracer and create custom repositories on the host.
+
+    This fixture automatically configures IPv6 support based on the host's network type and creates
+    version-appropriate repositories.
+
+    :param rex_contenthost: Remote execution enabled content host
+    :param target_sat: Target Satellite server
+    :return: ContentHost with tracer tools installed and configured
+    :raises ValueError: If RHEL version is not supported (7, 8, 9, or 10)
+    """
+    # add IPv6 proxy for IPv6 communication based on network type
+    if not rex_contenthost.network_type.has_ipv4:
+        rex_contenthost.enable_ipv6_dnf_and_rhsm_proxy()
+        rex_contenthost.enable_ipv6_system_proxy()
+
     # create a custom, rhel version-specific OS repo
     rhelver = rex_contenthost.os_version.major
+
+    # Support for RHEL 7, 8, 9, and 10
+    if rhelver not in settings.supportability.content_hosts.rhel.versions:
+        raise ValueError(
+            f'Unsupported RHEL version: {rhelver}. Supported versions: {settings.supportability.content_hosts.rhel.versions}'
+        )
     if rhelver > 7:
+        # RHEL 8, 9 and 10 use the same repository structure
         rex_contenthost.create_custom_repos(**settings.repos[f'rhel{rhelver}_os'])
     else:
+        # RHEL 7 has different repository structure
```

Comment:
Since it’s pretty well known in our framework now, I’d say we can skip the comments, and same above?